test/dynamo/test_misc.py

```diff
@@ -245,6 +245,29 @@ def compare_shapes(a, b, to_list):
             self, lambda a, b: compare_shapes(a, b, to_list=False), 2
         )
 
+
+    @patch.object(torch._dynamo.config, "dynamic_shapes", True)
+    def test_compare_shapes_with_constant(self):
+        def compare_shapes(a):
+            x = a.shape
+            if x[0] != 3:
+                return a * 4
+            return a * 3
+
+        guard_failure = None
+
+        def guard_failures(failure):
+            nonlocal guard_failure
+            guard_failure = failure
+
+        opt_fn = torch._dynamo.optimize(
+            "eager", nopython=True, guard_fail_fn=guard_failures
+        )(compare_shapes)
+        opt_fn(torch.randn([3, 4]))
+        opt_fn(torch.randn([4, 3]))
+        self.assertEqual(guard_failure.reason, "a.size()[0] == 3")
```

Comment:
Not sure what this test has to do with the PR

Reply:
Nothing, will give its own PR

Reply:
eh too much hassle, I figured you dont mind